aws-java-sdk-core/src/main/java/com/amazonaws/retry/PredefinedBackoffStrategies.java

```diff
@@ -103,7 +103,7 @@ public long delayBeforeNextRetry(AmazonWebServiceRequest originalRequest,
                                          AmazonClientException exception,
                                          int retriesAttempted) {
             return (retriesAttempted > MAX_RETRIES) ? maxBackoffTime :
-                    Math.min(((1 << retriesAttempted) * baseDelay), maxBackoffTime);
+                    Math.min(((1L << retriesAttempted) * baseDelay), maxBackoffTime);
```

Comment:
Thanks for the PR - good catch. I think in terms of implementation though we're potentially just dodging the problem - as soon as you get MAX_RETRIES >= 63 we're going to have the same issue (ie: Long's overflow) - we might be better adding another call to Math to help us. This should resolve the issue:

`Math.max(1, 1 << retriesAttempted)`

Also we should use the same with all the pre-canned back-off strategies.